### ISSUE TITLE
Running detector test with imagenet1k.data

### ISSUE THREAD

**Issue by ysabri** [pjreddie/darknet#232]
So whenever I run any of the ImageNet pre-trained models using the detector I get no predictions out at all. They all work using the classifier fine.  

Here is the command:
`./darknet detector test cfg/imagenet1k.data cfg/densenet201.cfg weights/densenet201.weights`

Here are the first few lines of my Makefile:
`GPU=0
CUDNN=0
OPENCV=1
OPENMP=0
DEBUG=0
`

**Comment by corentin87** [comment]
It's because you re missing the argument filename 
_test_detector(datacfg, cfg, weights, filename, thresh, hier_thresh);_

**Comment by ahsan856jalal** [comment]
They all are classifiers except yolo. Open the cfg and see the last two layer, if you want to make them detector first Remove last 4 layers from the cfg file i.e convolutional with linear activation, avgpool,softmax and cost. And add 2 layers : convolutional layer (linear activation) with filters= (# of classes+5)*#anchors. And second layer is region layer as in yolo.cfg (hint take these two layers from yolo.2.0.voc.cg file
Then try detector
